Scripts/Runtime/Entities/TaskDriver/TaskSet/TaskData/DataStream/DataSource/Data/AbstractData.cs

```diff
@@ -29,24 +29,39 @@ public static DataTargetID GenerateWorldUniqueID(IDataOwner dataOwner, Type abst
 
         public IDataOwner DataOwner { get; }
 
-        public AbstractData PendingCancelActiveData { get; }
+        public AbstractData ActiveCancelData { get; }
+
+        /// <summary>
+        /// The version of the data. This value is incremented each time write access is granted to the data allowing
+        /// consumers to check whether the data may have changed since the last time the consumer read it.
+        /// </summary>
+        /// <remarks>
+        /// Store this value at the time of read access and on next potential read use <see cref="IsDataInvalidated"/>
+        /// to check whether the data has potentially changed. This is the safest option as it accounts for other potential
+        /// conditions in the data.
+        ///
+        /// If comparing version numbers directly make sure to always use an equality check. The version number can
+        /// theoretically wrap and overflow so it's possible for the current value to be less than the previous value.
+        /// </remarks>
+        public uint Version { get; private set; }
```

Comment:
Not necessary for this PR but I wonder if we should wrap the `uint` into a custom struct that we only implement the equality check for and not the greater/less than checks. 

That would make it safer to use. Future TODO/PR of course though.